### PR TITLE
Running React frontend in production

### PR DESCRIPTION
**Pull request by davish**
This PR allows a dokku-deployed single-page react app to be run in production with a django backend. The main gripe I see people having with this is that django is serving a static file, `index.html`. This will be a temporary measure while the app is still hosted on Dokku. There's not a good way without over-riding the nginx config, which will mess with a bunch of stuff.

Haven't yet tested in production, but it works locally. The updates to the switchboard routing mean that when this gets deployed, the "under construction" page will still be shown on penncoursesearch.com, and the beta version of the site will be available on penncourseplan.com.

### CODE REVIEW

File: PennCourses/urls/pcp-construction.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from django.urls import path
+from django.views.generic import TemplateView
+
+from PennCourses.urls.base import urlpatterns
+
+
+urlpatterns = [
+    path('', TemplateView.as_view(template_name='plan/index.html')),
```
Comment (ArmaanT) [inline PennCourses/urls/pcp-construction.py]: `plan/index.html` and `plan/build/index.html` refer to two completely different `plan` directories, but that's not obvious in the code. Can we change this one to be something like `plan_construction/index.html`?

File: app.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "scripts": {
     "dokku": {
-      "predeploy": "python manage.py migrate --noinput"
+      "predeploy": "python manage.py migrate --noinput && cd frontend/plan && npm run build"
```
[the remark this answers is not in the document]
Comment (ArmaanT) [reply app.json]: Followup comment: I've got absolutely no idea how buildbacks work, so maybe an install isn't needed, but we should keep that in mind for the eventual transition to docker.

File: plan/urls.py [hunk 3]
```diff
@@ -8,8 +8,8 @@
 
 urlpatterns = [
     # omit semester parameter, since PCP only cares about the current semester.
-    path('', TemplateView.as_view(template_name='plan/index.html')),
     path('courses/', CourseListSearch.as_view()),
     path('courses/<slug:full_code>/', CourseDetailSearch.as_view()),
     path('requirements/', RequirementList.as_view()),
+    path('', TemplateView.as_view(template_name='plan/build/index.html')),
```
Comment (ArmaanT) [inline plan/urls.py]: See above comment on the `plan` directory confusion.

File: app.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (davish) [reply app.json]: Good points on both of those comments. I'm pretty sure we'll need to manually specify that we want to use multiple buildpacks, and yeah, I also do need to run npm install.

**Pull request merged**